❄️ Trivy-scan for our container fails

**Which jobs are flaking:**
[Trivy-Scan GHA](https://github.com/eitco/cluster-api-addon-provider-cdk8s/blob/main/.github/workflows/scan.yml)

**Which tests are flaking:**
https://github.com/eitco/cluster-api-addon-provider-cdk8s/actions/runs/16417609909/job/46387309466

**Testgrid link:**

**Reason for failure (if possible):**

```
make[1]: Leaving directory '/home/runner/work/cluster-api-addon-provider-cdk8s/cluster-api-addon-provider-cdk8s'
2025-07-21T13:00:36.009Z	FATAL	image scan error: scan error: unable to initialize a scanner: unable to initialize a docker scanner: 4 errors occurred:
	* unable to inspect the image (ghcr.io/eitco/cluster-api-addon-provider-cdk8s/cluster-api-cdk8s-controller-amd64:dev): Error: No such image: ghcr.io/eitco/cluster-api-addon-provider-cdk8s/cluster-api-cdk8s-controller-amd64:dev
	* unable to initialize Podman client: no podman socket found: stat /run/user/1001/podman/podman.sock: no such file or directory
	* failed to initialize a containerd client: failed to dial "/run/containerd/containerd.sock": connection error: desc = "transport: error while dialing: dial unix /run/containerd/containerd.sock: connect: permission denied"
	* GET https://ghcr.io/v2/eitco/cluster-api-addon-provider-cdk8s/cluster-api-cdk8s-controller-amd64/manifests/dev: MANIFEST_UNKNOWN: manifest unknown



./hack/verify-container-images.sh: line 67: R1: unbound variable
make: *** [Makefile:385: verify-container-images] Error 1
```

**Anything else we need to know:**
- links to go.k8s.io/triage appreciated
- links to specific failures in spyglass appreciated

--- a/Dockerfile
+++ b/Dockerfile
@@ -77,8 +77,8 @@ ARG ARCH
 # Set shell with pipefail option for better error handling
 SHELL ["/bin/sh", "-o", "pipefail", "-c"]
 
-RUN apk add --no-cache ca-certificates=20231108 curl=8.1.2-r0 nodejs=18.16.0-r0 npm=9.5.1-r0 \
-    && npm install -g cdk8s-cli@2.200.109 \
+RUN apk add --no-cache ca-certificates=20250619-r0 curl=8.12.1-r0 nodejs=20.15.1-r0 npm=10.9.1-r0 \
+    && npm install -g cdk8s-cli@2.200.152 \
     && curl -fsSL -o go1.24.4.linux-${ARCH}.tar.gz https://go.dev/dl/go1.24.4.linux-${ARCH}.tar.gz \
     && tar -C /usr/local -xzf go1.24.4.linux-${ARCH}.tar.gz \
     && rm go1.24.4.linux-${ARCH}.tar.gz \

